src/manager.cpp

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "manager.hpp"
+
+#include <nlohmann/json.hpp>
+#include <phosphor-logging/lg2.hpp>
+
+#include <exception>
+#include <fstream>
+#include <iterator>
+#include <ranges>
+#include <string>
+#include <string_view>
+
+namespace data_sync
+{
+
+Manager::Manager(const fs::path& dataSyncCfgDir)
+{
+    parseConfiguration(dataSyncCfgDir);
```

Comment:
I believe this could be converted into an async call, but it can be addressed separately based on the requirements.